app/controllers/comments/delete.js

```diff
@@ -0,0 +1,88 @@
+import { db } from "~/utils/db.server";
+import generateSessionIdHash from "~/utils/backend/crypto";
+import { generateMinMaxDates } from "~/utils/backend/comments";
+import { deleteCommentSchema } from "~/utils/backend/validators/comments";
+import { DEFAULT_ERROR_MESSAGE } from "~/utils/backend/constants";
+import {
+  INVALID_PARAMS_FOR_OPERATION_ERROR_MESSAGE,
+  DELETE_COMMENT_ERROR_MESSAGE,
+} from "~/utils/constants";
+
+export const deleteComment = async (body) => {
+  const { error, value } = deleteCommentSchema.validate(body);
+
+  if (error) {
+    return {
+      errors: [
+        {
+          message: INVALID_PARAMS_FOR_OPERATION_ERROR_MESSAGE,
+          detail: error,
+        },
+      ],
+    };
+  }
+
+  const { commentId: id, accessToken, userEmail } = value;
+
+  const sessionHash = generateSessionIdHash(accessToken, id);
+  const { minDate, maxDate } = generateMinMaxDates();
+
+  const deleteCommentResponse = await db.Comments.deleteMany({
+    where: {
+      id,
+      OR: [
+        {
+          userEmail,
+        },
+        {
+          AND: [
+            {
+              sessionHash,
+            },
+            {
+              createdAt: { lte: new Date(maxDate) },
+            },
+            {
+              createdAt: { gte: new Date(minDate) },
+            },
+          ],
+        },
+      ],
+    },
+  });
+
+  if (
+    deleteCommentResponse.count === undefined ||
+    typeof deleteCommentResponse.count !== "number"
+  ) {
+    return {
+      error: {
+        message: DEFAULT_ERROR_MESSAGE,
+        detail: "Something went wrong trying to delete the comment",
+      },
+    };
+  }
+
+  if (deleteCommentResponse.count === 0) {
+    return {
+      error: {
+        message: DELETE_COMMENT_ERROR_MESSAGE,
+        detail:
+          "Comment not found or user does not have deletion rights over the comment",
```

Comment:
We can add the comment id here for debugging

Reply:
Got it, comment id added!